src/cirrus/release.py

```diff
@@ -517,91 +529,97 @@ def merge_release(opts):
 
         # merge release branch into master
         LOGGER.info("Tagging and pushing {0}".format(tag))
-
-        sha = ghc.repo.head.ref.commit.hexsha
-
-        if rel_conf['update_github_context']:
-            LOGGER.info("Setting {} for {}".format(
-                rel_conf['github_context_string'],
-                sha)
-            )
-            ghc.set_branch_state(
-                'success',
-                rel_conf['github_context_string'],
-                branch=sha
-            )
-        if rel_conf['wait_on_ci']:
-            #
-            # wait on release branch CI success
-            #
-            LOGGER.info("Waiting on CI build for {0}".format(release_branch))
-            ghc.wait_on_gh_status(
-                sha,
-                timeout=rel_conf['wait_on_ci_timeout'],
-                interval=rel_conf['wait_on_ci_interval']
+        if opts.skip_master:
+            LOGGER.info('Skipping merging to {}'.format(master))
+        if opts.skip_develop:
+            LOGGER.info('Skipping merging to {}'.format(develop))
+
+        if not opts.skip_master:
+            sha = ghc.repo.head.ref.commit.hexsha
+
+            if rel_conf['update_github_context']:
+                LOGGER.info("Setting {} for {}".format(
+                    rel_conf['github_context_string'],
+                    sha)
+                )
+                ghc.set_branch_state(
+                    'success',
+                    rel_conf['github_context_string'],
+                    branch=sha
+                )
+            if rel_conf['wait_on_ci']:
+                #
+                # wait on release branch CI success
+                #
+                LOGGER.info("Waiting on CI build for {0}".format(release_branch))
+                ghc.wait_on_gh_status(
+                    sha,
+                    timeout=rel_conf['wait_on_ci_timeout'],
+                    interval=rel_conf['wait_on_ci_interval']
+                )
+
+            LOGGER.info("Merging {} into {}".format(release_branch, master))
+            ghc.pull_branch(master)
+            ghc.merge_branch(release_branch)
+            sha = ghc.repo.head.ref.commit.hexsha
+
+            if rel_conf['update_github_context']:
+                LOGGER.info("Setting {} for {}".format(
```

Comment:
You might want to use unicode strings if we're using substrings from e.g. github responses